app/services/IncomeSourceDetailsService.scala

```diff
@@ -224,5 +205,30 @@ class IncomeSourceDetailsService @Inject()(val incomeTaxViewChangeConnector: Inc
       }
     }.toEither
   }
+
+  private def getCeasedBusinesses(sources: IncomeSourceDetailsModel): List[CeasedBusinessDetailsViewModel] = {
```

Comment:
we usually return Either[Throwable, ViewModel] from the service, and log that error in the the caller, which is usually controller